### PR TITLE
Add report when a job was finished

### PR DESCRIPTION
**Pull request by cwen0**
@zhouqiang-cl PTAL

### CODE REVIEW

File: oasis/job.py [hunk 1]
```diff
@@ -34,12 +41,14 @@ def run(self):
...
         for md in self.models:
             model = Models[md.get("name")]
-            model_runner = model(self.id, md, self.data_source, self.slack_channel)
+            model_runner = model(self.id, md, self.data_source, self.slack_channel, self.timeout)
             t = Thread(target=model_runner.run())
             t.start()
             self.running_models.append(model_runner)
             self.threads[md.get("name")] = t
 
+        self.timer = Timer(timeparse(self.timeout), self.timeout_action)
```
Comment (zhouqiang-cl) [inline oasis/job.py]: what happend if run many jobs. then timer will change

Comment (zhouqiang-cl) [reply oasis/job.py]: so, One job just run once?

Comment (cwen0) [reply oasis/job.py]: yes, One job only run once.

File: oasis/models/model.py [hunk 2]
```diff
@@ -71,3 +75,29 @@ def _parse_data(self, data):
...
+        }
+
+
+class ModelReport(object):
+    def __init__(self, model_name, job_id, model_config):
+        self.job_id = job_id
+        self.model_name = model_name
+        self.model_config = model_config
+        self.start_time = time.asctime(time.localtime(time.time()))
+        self.metrics_report = dict()
```
Comment (zhouqiang-cl) [inline oasis/models/model.py]: can we have such as how many points already analysis。 I did not find it

File: oasis/models/rules.py [hunk 3]
```diff
@@ -65,7 +76,21 @@ def compute(self, metric, query_expr, config, rules):
                             "start to match with rule"
                             .format(log_prefix=self.log_prefix,
                                     metric=metric, value=features_value))
-                self.match_rules(metric, features_value, rules)
+                with self.lock:
+                    self.report.metrics_report[metric]["predict_count"] = \
+                        self.report.metrics_report[metric]["predict_count"] + 1
+
+                is_match, not_match_rule = self.match_rules(metric, features_value, rules)
```
Comment (zhouqiang-cl) [inline oasis/models/rules.py]: if match, where is the report

Comment (cwen0) [reply oasis/models/rules.py]: `match` is normal, so we not need report.

File: oasis/models/rules.py [hunk 4]
```diff
@@ -65,7 +76,21 @@ def compute(self, metric, query_expr, config, rules):
                             "start to match with rule"
                             .format(log_prefix=self.log_prefix,
                                     metric=metric, value=features_value))
-                self.match_rules(metric, features_value, rules)
+                with self.lock:
+                    self.report.metrics_report[metric]["predict_count"] = \
```
Comment (zhouqiang-cl) [inline oasis/models/rules.py]: predict_count + len(data_set)?

**Pull request merged**